spirit/core/utils/markdown/block.py

```diff
@@ -41,12 +41,22 @@ class BlockGrammar(mistune.BlockGrammar):
     # * https://www.youtube.com/watch?v=Z0UISCEe52Y
     # * http://youtu.be/afyK1HSFfgw
     # * https://www.youtube.com/embed/vsF0K3Ou1v0
+    #
+    # Also works for timestamps:
+    # * https://www.youtube.com/watch?v=Z0UISCEe52Y&t=1m30s
+    # * https://www.youtube.com/watch?v=O1QQajfobPw&t=1h1m38s
+    # * https://www.youtube.com/watch?v=O1QQajfobPw&feature=youtu.be&t=3698
+    # * https://youtu.be/O1QQajfobPw?t=3698
+    # * https://youtu.be/O1QQajfobPw?t=1h1m38s
+    #
     youtube = re.compile(
         r'^https?://(www\.)?'
         r'(youtube\.com/watch\?v='
         r'|youtu\.be/'
         r'|youtube\.com/embed/)'
         r'(?P<id>[a-zA-Z0-9_\-]{11})'
+        r'((&|\?)?feature=youtu.be)?'
+        r'((&|\?)?t=(?P<start_hours>[0-9]+h)?(?P<start_minutes>[0-9]+m)?(?P<start_seconds>[0-9]+s?)?)?'
```

Comment:
* The & or \? should not be optional.
* All matches should be limited. Seconds should be limited to 5 digits, minutes and hours to 2. Although, I'm not sure if minutes can be something higher than 59.

There is a bigger issue with this regex, the params must be in exactly that order to work, ie: this `https://www.youtube.com/watch?v=O1QQajfobPw&t=3698&feature=youtu.be` won't work coz `feature=` comes after `t=`. But it's not a blocker, I can fix it later.